scripts/ipr-check.js

```diff
@@ -0,0 +1,179 @@
+			if (res.statusCode < 200 || res.statusCode >= 300) {
+				return reject(new Error(`Status Code: ${res.statusCode}; ${url}`));
+			}
+
+			const data = [];
+
+			res.on('data', chunk => {
+				data.push(chunk);
+			});
+
+			res.on('end', () => resolve(String(Buffer.concat(data))));
+		});
+
+		req.on('error', reject);
+
+		if (postData) {
+			req.write(postData);
+		}
+
+		req.end();
+	});
+};
+
+const perPage = 100;
+
+function getAuthorFromCommit(commitObj) {
+	if (!commitObj) {
+		return false;
+	}
+	const { author } = commitObj;
+	if (!author) {
+    throw new Error("Missing author for commit " + commitObj.sha);
+	}
+	return author.login;
+}
+
+async function getAllCommits(page = 1) {
+	const commitsURL = `https://api.github.com/repos/${slug}/commits?anon=1&per_page=${perPage}&page=${page}&sha=${sha}`;
+	const commits = await request(commitsURL).then((json) => JSON.parse(json));
+	return [...new Set([].concat(
+		commits.flatMap(x => getAuthorFromCommit(x) || []),
+		commits.length < perPage ? [] : await getAllCommits(page + 1),
+	))];
+}
+
+const authors = getAllCommits().then((authors) => {
+	const knowns = authors.filter(x => typeof x === 'string');
+	console.log(`Found ${knowns.length} authors: ${knowns.join(',')}\n`);
+	return knowns;
+});
+
+const teamURL = (team) => `https://api.github.com/orgs/tc39/teams/${team}`;
+
+function getMembers(teamID, page = 1) {
+	const memberURL = `https://api.github.com/teams/${teamID}/members?per_page=100&page=${page}`;
+	const data = request(memberURL).then((json) => JSON.parse(json));
+	return data.then((data) => {
+		if (data.length === 0) {
+			return data;
+		}
+		return getMembers(teamID, page + 1).then(nextPage => {
+			return data.concat(nextPage);
+		});
+	});
+}
+
+function handler(kind) {
+	return (data) => {
+		const names = new Set(data.map(x => x.login.toLowerCase()));
+		console.log(`Found ${names.size} ${kind}: ${[...names].join(',')}\n`);
+		return names;
+	}
+}
+
+const delegates = request(teamURL('delegates')).then((json) => JSON.parse(json)).then(data => {
+	return getMembers(data.id);
+}).then(handler('delegates'));
+
+const emeriti = request(teamURL('emeriti')).then((json) => JSON.parse(json)).then(data => {
+	return getMembers(data.id);
+}).then(handler('emeriti'));
+
+const usernames = request(sheetData).then((json) => JSON.parse(json)).then(data => {
+	if (!Array.isArray(data.values)) {
+		throw 'invalid data';
+	}
+	const usernames = new Set(
+		data.values
+			.flat(1)
+			.map(x => x.replace(/^(https?:\/\/)?github\.com\//, '').replace(/^@/, '').toLowerCase())
+			.filter(x => /^[a-z0-9_-]{1,39}$/gi.test(x))
+			.sort((a, b) => a.localeCompare(b))
+	);
+	console.log(`Found ${usernames.size} usernames: ` + [...usernames].join(',') + '\n');
+	return usernames;
+});
+
+const exceptions = new Set([
+  'jaro-sevcik', // Google empolyee
+  'jkrems', // Google employee
+  'josephschorr', // former Google empolyee
+	'sideshowbarker', // Mozilla employee
```

Comment:
These are people that are not in any delegates group in the tc39 org, but work (or used to work at the time of their contribution) for a company that is part of ECMA.

Reply:
Can confirm for @jkrems.